cmd/boulder-ca/main.go

```diff
@@ -193,6 +195,17 @@ func main() {
 		cmd.FailOnError(err, "Failed to load CT Log List")
 	}
 
+	// Check that the system clock believes it is synced with a reliable time
+	// server. This is important for the CA because it selects our notBefore and
+	// notAfter values, which MUST NOT be more than 48 hours from the actual time
+	// of the signing operation per BRs Section 7.1.2.7.
+	var timex = new(unix.Timex)
+	status, err := unix.Adjtimex(timex)
+	cmd.FailOnError(err, "Failed to check system time")
+	if status == unix.TIME_ERROR {
```

Comment:
I think we should check if status & STA_UNSYNC is set

Reply:
Per https://man7.org/linux/man-pages/man2/adjtimex.2.html, the status will be TIME_ERROR if STA_UNSYNC is set.

Or do you mean only error out if both are set, i.e. not failing in the other cases that also can cause a status of TIME_ERROR?

Reply:
Never mind, I think you're good.  I was a bit confused about what `status` was - I thought it was the `status` field in the timex buffer, which is a bitmask.  But the return value is different.

Reply:
Here's a small test program I wrote to convince myself this is OK:

```
package main


import "golang.org/x/sys/unix"
import "fmt"

func main() {
    var timex = new(unix.Timex)
    status, err := unix.Adjtimex(timex)
    fmt.Println(status, err)
    fmt.Println(unix.TIME_ERROR)
    fmt.Println("status == unix.TIME_ERROR", status == unix.TIME_ERROR)
    fmt.Println(timex.Status)
}
```

and the output:

```
time-test
5 <nil>
5
status == unix.TIME_ERROR true
64
```

Reply:
I'll name the variable something more descriptive.